src/Abp.AutoMapper/AutoMapper/AutoMapAttribute.cs

```diff
@@ -19,11 +19,8 @@ public override void CreateMap(IMapperConfigurationExpression configuration, Typ
                 return;
             }
 
-            foreach (var targetType in TargetTypes)
-            {
-                configuration.CreateMap(type, targetType, MemberList.Source);
-                configuration.CreateMap(targetType, type, MemberList.Destination);
-            }
+            configuration.CreateAutoAttributeMaps(type, TargetTypes, MemberList.Source);
+            configuration.CreateAutoAttributeMaps(type, TargetTypes, MemberList.Destination);
```

Comment:
Hi @naazz1992,

I have found something changing the existing behavior. If you look at the second line in foreach loop, it creates a mapping like this;

`configuration.CreateMap(targetType, type, MemberList.Destination);`

But in your version, both lines creates mapping from **type -> TargetTypes** which changes the current behavior.

If you add a test like the one below, it will fail.

```c#
[Fact]
public void Should_Map_Two_Way_When_AutoMAp_Attribute_Is_Used()
{
	MyClass3 obj2 = new MyClass3
	{
		TestProp = "test",
		AnotherValue = 1
	};

	var obj1 = _mapper.Map<MyClass1>(obj2);

	obj1.TestProp.ShouldBe("test");
}
```
I think you should change mapping creation in AutoMapAttribute like below. What do you think ?

```c#
configuration.CreateAutoAttributeMaps(type, TargetTypes, MemberList.Source);

foreach (var targetType in TargetTypes)
{
	configuration.CreateAutoAttributeMaps(targetType, new[] { type }, MemberList.Destination);
}
```